tests/epyccel/test_arrays.py

```diff
@@ -3991,12 +3994,12 @@ def test_numpy_arange_with_dtype(language):
 def test_numpy_arange_negative_step(language):
     f1 = arrays.arr_arange_5
     f2 = epyccel(f1, language = language)
-    np.testing.assert_array_almost_equal(f1(), f2(), decimal = 9)
+    np.testing.assert_allclose(f1(), f2())
 
 def test_numpy_arange_negative_step_2(language):
     f1 = arrays.arr_arange_6
     f2 = epyccel(f1, language = language)
-    np.testing.assert_array_almost_equal(f1(), f2(), decimal = 9)
+    np.testing.assert_allclose(f1(), f2())
```

Comment:
You should also check the return type (maybe this explains your precision problems). If the type is wrong then this problem might be linked to an issue caught by @aihya and fixed in PR #1169

Reply:
If i understand correctly what @aihya explained to me is that the error could occur when converting an `np.array` to a python array the precision may go down in some cases, but i'm still not entirely sure about the implications of this.

Reply:
@Pinkyboi Did you update the tests?

Reply:
@EmilyBourne i did revert back some changes there since i noticed that the problem was caused by my implementation, i left the test on the return type cuz i thought i may be useful still.